VueJS/index.js

```diff
@@ -1,13 +1,35 @@
 import List from './components/List.js'
 import ListItem from './components/ListItem.js'
 
-new Vue({
+var myApp = new Vue({
   el: '#app',
+  methods: {
+	addTodo: function(){
+		this.todos.push(
+			{ id: this.getNextId(), message: '', }
+		);
+	},
+	saveTodo: function(todo){
+		this.todos[todo.index].message = todo.message;
+	},
+	deleteTodo: function(index){
+		this.todos.splice(index, 1);
```

Comment:
`splice` has the side effect of mutating the array? Urgh. To a new developer it's not that obvious that `this.todos` gets mutated here. This is why I prefer `filter`, `map`, `find` and friends. I see no better solution here, though.

I found: https://jaketrent.com/post/remove-array-element-without-mutating/ but this bloats the code unnecessarily.